Fixed clang configuration file

Hadatko opened the pull request:
+ reformatted code.

# Pull request

**Choose Correct**

- [ ] bug
- [X] feature

**Describe the pull request**
<!--
A clear and concise description of what the pull request is.
-->
I noticed that some clang-format config values were changed for version we have. So i adapt them and added one more which was not available.

**To Reproduce**
<!--
Steps to reproduce the behavior.
-->

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

**Screenshots**
<!--
If applicable, add screenshots to help explain your problem.
-->

**Desktop (please complete the following information):**

- OS<!--[e.g. iOS]-->:
- eRPC Version<!--[e.g. v1.9.0]-->:

**Steps you didn't forgot to do**

- [X] I checked if other PR isn't solving this issue.
- [X] I read Contribution details and did appropriate actions.
- [X] PR code is tested.
- [X] PR code is formatted.
- [X] Allow edits from maintainers pull request option is set (recommended).

**Additional context**
<!--
Add any other context about the problem here.
-->

MichalPrincNXP commented:
Hi Dusan, thank you. Would it be possible to move to LLVM16 to be aligned with MCUXpressoSDK formatting?

Hadatko commented on .clang-format:
```diff
@@ -45,27 +46,35 @@ Cpp11BracedListStyle: false
-SortIncludes: true
-SortUsingDeclarations: true
-SpacesInAngles: false
+SortIncludes: "CaseSensitive"
```
Actually i would like to keep it, but restructualize includes in files. If you want keep any include in specific place we just need insert newline before/after that particular case. Value i choosed is based on previous true value. I want do something similar based on my current experience
first header file of declaration
<space>
<erpc includes>
<space>
<nxp includes>
<space>
<freertos includes>
<space>
<other includes>

Hadatko commented:
Hi @isolansky , if we want to be consistent we can take a look how it looks like in initializing contructor. Newline is before comma. Then we need change both.

![image](https://github.com/EmbeddedRPC/erpc/assets/13130700/9dc67efd-80df-4369-a2d8-7bb6d7e552f5)

isolansky commented:
From my understanding, goal is to have code, easily mergable with other projects (under MCUX) or using MSDK in other way.
From my experiences I always got commas at the line end from OSS either from in-company code. So thus asking if there is any other kind of strong motivation for this. If it is just personal feeling, or some big related project.

Hadatko commented:
I think you are mixing apples and pears ;) We are trying to align C++ code style with NXP code style. But NXP doesn't have C++ code style (afaik) so we should discuss what is best for C++ specific cases. My suggestion looks like block so easier to read.

Anyway as mentioned before. If we do your requested change it would lead to inconsistency with constructor declaration. So we need keep current settings or change both.

(it looks this type of inconsistency is present in webkit. Do you still like Mozilla? They are using my proposed change :D)

![image](https://github.com/EmbeddedRPC/erpc/assets/13130700/9934c735-43d3-4448-bee7-4312c7a170e7)
![image](https://github.com/EmbeddedRPC/erpc/assets/13130700/e5dd7d96-cee3-4bb0-8cc6-d0140245cb34)
![image](https://github.com/EmbeddedRPC/erpc/assets/13130700/16367eaa-3bc5-4125-8bc0-1a821cfea9e0)

Hadatko commented:
@MichalPrincNXP @isolansky Your requests are placed. I noticed that clang check was not call on PR change so i added this option and moved excluded files list to its own file so we can have one source of excluded files for two different usage of clang format.

isolansky commented:
Thank you, also for point out that include sorting based on grouping. This will be subject of future consideration.

Hadatko commented:
@isolansky That sounds great. Also in that case you would have to use:
IncludeBlocks: Preserve

Hadatko commented:
@MichalPrincNXP this should be ready too.

MichalPrincNXP commented:
thank you, Dusan
